src/components/coin-row/FastTransactionCoinRow.tsx

```diff
@@ -0,0 +1,156 @@
+  native: any;
+  status: keyof typeof TransactionStatusTypes;
+  type: keyof typeof TransactionTypes;
+  theme: ThemeContextProps;
+}) {
+  const { colors } = theme;
+  const isFailed = status === TransactionStatusTypes.failed;
+  const isReceived =
+    status === TransactionStatusTypes.received ||
+    status === TransactionStatusTypes.purchased;
+  const isSent = status === TransactionStatusTypes.sent;
+
+  const isOutgoingSwap = status === TransactionStatusTypes.swapped;
+  const isIncomingSwap =
+    status === TransactionStatusTypes.received &&
+    type === TransactionTypes.trade;
+
+  let coinNameColor = colors.dark;
+  if (isOutgoingSwap) coinNameColor = colors.blueGreyDark50;
+
+  let balanceTextColor = colors.blueGreyDark50;
+  if (isReceived) balanceTextColor = colors.green;
+  if (isSent) balanceTextColor = colors.dark;
+  if (isIncomingSwap) balanceTextColor = colors.swapPurple;
+  if (isOutgoingSwap) balanceTextColor = colors.dark;
+
+  const nativeDisplay = native?.display;
+  const balanceText = nativeDisplay
+    ? [isFailed || isSent ? '-' : null, nativeDisplay].filter(Boolean).join(' ')
+    : '';
+
+  return (
+    <View style={cx.bottomRow}>
+      <Text color={{ custom: coinNameColor || colors.dark }} size="16px">
+        {description}
+      </Text>
+      <Text
+        align="right"
+        color={{ custom: balanceTextColor ?? colors.dark }}
+        size="16px"
+        weight={isReceived ? 'medium' : undefined}
+      >
+        {balanceText}
+      </Text>
+    </View>
+  );
+});
+
+export default React.memo(function TransactionCoinRow({
+  item,
+  theme,
+  onTransactionPress,
+}: {
+  item: any;
+  theme: ThemeContextProps;
+  onTransactionPress: (item: any) => void;
+}) {
+  const { mainnetAddress } = item;
+  const { colors } = theme;
+  const handleTransactionPress = useCallback(() => onTransactionPress(item), [
+    item,
+    onTransactionPress,
+  ]);
+
+  return (
+    <ButtonPressAnimation onPress={handleTransactionPress} scaleTo={0.96}>
+      <View style={cx.wholeRow}>
+        <View style={cx.icon}>
+          <FastCoinIcon
+            address={mainnetAddress || item.address}
+            assetType={item.assetType}
+            symbol={item.symbol}
+            theme={theme}
+          />
+        </View>
+        <View style={cx.column}>
+          <View style={cx.topRow}>
+            <FastTransactionStatusBadge
+              colors={colors}
+              pending={item.pending}
+              status={item.status}
+              title={item.title}
+            />
+            <View style={cx.balance}>
+              <Text
+                color={{ custom: colors.alpha(colors.blueGreyDark, 0.5) }}
+                numberOfLines={1}
+                size="14px"
+              >
+                {item.balance?.display ?? ''}
+              </Text>
+            </View>
+          </View>
+          <BottomRow
+            description={item.description}
+            native={item.native}
```

Comment:
i guess we can pass `item.native.display` directly here instead so we won't pass an object